packages/xchain-ethereum/src/etherscan-api.ts

```diff
@@ -78,7 +78,14 @@ export const getETHTransactionHistory = async ({
   if (endblock) url += `&endblock=${endblock}`
 
   const ethTransactions: ETHTransactionInfo[] = await axios.get(url).then((response) => response.data.result)
-  return ethTransactions.filter((tx) => !bn(tx.value).isZero()).map(getTxFromEthTransaction)
+  const filterTxs = ethTransactions.filter((tx) => tx.from !== tx.to)
+  let selfTxs = ethTransactions.filter((tx) => tx.from === tx.to)
+  while (selfTxs.length) {
+    const selfTx = selfTxs[0]
+    filterTxs.push(selfTx)
+    selfTxs = selfTxs.filter((tx) => tx.hash !== selfTx.hash)
+  }
```

Comment:
Pls extract filter function + write a test. Use same filter function in `getTokenTransactionHistory` (as youi have already done it)